modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientsManagerTests.java

```diff
@@ -203,12 +204,15 @@ public void testClientsLifeCycleForSingleProject() throws Exception {
         }
         assertClientNotFound(projectId, clientName);
 
-        assertBusy(() -> assertTrue(clientsHolder.isClosed()));
-        final var e = expectThrows(
-            IllegalStateException.class,
-            () -> clientsHolder.client(createRepositoryMetadata(randomFrom(clientName, anotherClientName)))
-        );
-        assertThat(e.getMessage(), containsString("Project [" + projectId + "] clients holder is closed"));
+        assertBusy(() -> {
+            assertTrue(clientsHolder.isClosed());
+            final var e = expectThrows(AlreadyClosedException.class, () -> {
+                var client = clientsHolder.client(createRepositoryMetadata(randomFrom(clientName, anotherClientName)));
+                client.decRef();
+            });
+            assertThat(e.getMessage(), containsString("Project [" + projectId + "] clients holder is closed"));
```

Comment:
nit: doing all of this inside the `assertBusy()` means that we will still pass the test if we temporarily throw a different exception during shutdown. It'd be better if we just looped until we see _some_ exception, and then asserted that the first-thrown exception is an `AlreadyClosedException` with the right message.